feat(view): introduce parameter metadata

xHeaven opened the pull request:
For future LSP support. Thoughts? Ideas for improvement?

innocenzi commented on packages/view/src/ViewComponentParameters.php:
```diff
@@ -0,0 +1,87 @@
+use IteratorAggregate;
+use JsonSerializable;
+
+final class ViewComponentParameters implements ArrayAccess, IteratorAggregate, Countable, JsonSerializable
```
We have an internal `ArrayInterface` that implements all these!

innocenzi reviewed (commented):
A few ideas:
- Maybe have this new thing return more metadata than just parameters (eg. a description for the component itself, for IDE insights)
- Maybe actually have a separate interface to avoid the breaking change/inconvenience (not sure about this)

xHeaven commented:
> A few ideas:
> 
> * Maybe have this new thing return more metadata than just parameters (eg. a description for the component itself, for IDE insights)
> * Maybe actually have a separate interface to avoid the breaking change/inconvenience (not sure about this)

I've added some more metadata methods, including ones for:
- description
- slots
- deprecation notices

I also extracted everything into a separate interface, so this feature can be opt-in (a pretty recommended one).

I'm all ears for any feedback or suggestions 👀
